client-src/rundump.c

```diff
@@ -172,15 +175,35 @@ main(
 # endif
 #endif
 
-
     /*
      * Build the array
-     */
-
-    g_ptr_array_add(array, g_strdup(dump_program));
-
-    for (i = 1; argv[i]; i++)
-        g_ptr_array_add(array, quote_string(argv[i]));
+     */	 
+	g_ptr_array_add(array, g_strdup(dump_program));
+	
+	// No need to deallocate "prgName", as strrchr is not returning newly allocated memory
+	// but just returning an offset to the string.
+	prgName = g_strrstr(dump_program, "/");
+	if (prgName == NULL)
+	{
+		prgName = dump_program;
+	}
+	else
+	{
+		prgName++;
+	}
+	if (g_strcmp0(prgName, "dump") == 0) {
+		if (validate_dump_option(argc, argv) != 0) {
+			error(" ");
+		}
+	}
+	else if (g_strcmp0(prgName, "xfsdump") == 0) {
+		if (validate_xfsdump_options(argc, argv) != 0) {
+			error(" ");
+		}
+	}
+    for (i = 1; argv[i]; i++) {
+		g_ptr_array_add(array, quote_string(argv[i]));
+    }
```

Comment:
The whitespace in this chunk is inconsistent (probably tabs vs. spaces).

Reply:
Hello Dustin,
Thank you for your review comment.
Yes, I will make the whitespace consistent in newly added code.